Bryce phase 2

brycerandolph opened the pull request:
Needs team review so I can make necessary changes before merging with master

brycerandolph committed:
First take of the revised card, hand, and deck classes

I am sure there are a few errors in here, I also need to fix the nested
folders as well

apainintheneck commented:
A few small things.

1. Index++ should be after the if statement in Deck.removeCard(). Otherwise it is an off by one error.
2. The for each loop in Deck.addCard() should look like: for(Card inCard : cards).
3. The if statement in Deck.addCard() returns after comparing two cards. You can only know for sure that a card is already in the deck when the deck has been completely searched.
4. For Deck.addCard() and Deck.removeCard() you are not accounting for when there are multiple decks which means there can be duplicates of a single card.

apainintheneck committed:
Fixed Card.class.

Card.arraySort():
-Changed line 92: (arraySize - i) to (arraySize - i - 1)
-Changed line 96: int temp... to Card temp...

apainintheneck committed:
Deleted allValues array from Card class.

It is redundant with the valuRanks array.

apainintheneck committed:
Fixed small bugs in Hand class.

-line 73: Changed name of enum to be valid
-line 63: Changed how Card.arraySort method is accessed.

apainintheneck commented:
I think I've cleaned up the Card and Hand classes. I'll start working on the Deck class next.

apainintheneck committed:
Add changes to Deck class.

Changes:
-Fixed removeCard class to only remove card if found in the deck.
-Created private countInDeck() method to be used by the addCard()
method.

apainintheneck committed:
Changed Deck.addCard() method.

Now it uses Deck.countInDeck() to find out if the deck already contains
the maximum number of that card. If not and the deck is not full, add
the card to the top.

apainintheneck commented:
I think that I've now fixed the Deck class. Next, I will work on the tests.

apainintheneck committed:
Added Card.valuRanks[] to Deck class.

apainintheneck committed:
Added tests for the Deck, Hand, and Card classes.

apainintheneck commented:
I finished the tests. If it looks good, merge it in. Otherwise, I'll merge it when I get back from my walk.

apainintheneck committed:
Merge branch 'master' into Bryce-Phase-2

Conflicts:
	Phase-1-src/bin/Assig5.class
	Phase-1-src/src/Assig5.java

apainintheneck committed:
Resolving conflicts with Master.
